pydantic/_internal/_generate_schema.py

```diff
@@ -406,25 +395,10 @@ def _config_wrapper(self) -> ConfigWrapper:
     def _types_namespace(self) -> dict[str, Any] | None:
         return self._types_namespace_stack.tail
 
-    @property
-    def _current_generate_schema(self) -> GenerateSchema:
-        cls = self._config_wrapper.schema_generator or GenerateSchema
-        return cls.__from_parent(
-            self._config_wrapper_stack,
-            self._types_namespace_stack,
-            self.model_type_stack,
-            self._typevars_map,
-            self.defs,
-        )
```

Comment:
But I'm wondering, what was the purpose of creating new instances, if every attribute was being passed in?

Apart from `field_name_stack` where a new instance is created. By the way, doesn't that break anything?

Reply:
The purpose of creating new instances was to give the model a change to inject its custom core schema generator (some subclass of `GenerateSchema`

Reply:
In theory this can lead to inconsistent state, where in the process of generating a core schema for a field (and thus having one field name in the stack), we end up generating something else. However, the field names are only used when generating schema for `*Validator` classes, that can only be used when defining a field (meaning a new field name will be added to the stack, so it doesn't matter if the stack was reset or not).

Reply:
Hmm. Interesting. Not sure I follow entirely - happy to take a closer look together and potentially refactor that structure.

Reply:
For instance:

```python
class Sub(BaseModel):
    f: 'Forward'

Forward = int

class Model(BaseModel):
    sub: Sub
```

When `Sub` is defined, schema generation fails because `Forward` isn't defined. when `Model.sub` is being defined, the field name stack is `['sub']`. We try to get the core schema of `Sub` but it needs to be rebuilt. Previous to your change, we would create a new `GenerateSchema` instance with an empty field name stack. But now, we build the core schema instance of `Sub` with the same instance. When we try to build the core schema of `Sub.f`, the field name stack is `['sub', 'f']`, so it is still valid. It can be inconsistent if we try to build something _not_ related to a field in `Sub` (e.g. evaluating the type annotation of `__pydantic_extra__`, etc), but anyway the field name stack is only used by `*Validator` classes and such classes could only be used (in our example) in the context of field `f`, so the field name stack would be `['sub', 'f']` so all good.